ding_dibs.email_reports.inc

```diff
@@ -0,0 +1,292 @@
+        form_error($element, t('Please enter a valid email address.'));
+      }
+      elseif (empty($email)) {
+        form_error($element, t('Remove the empty string.'));
+      }
+    }
+  }
+  else {
+    form_error($element, t('Maximum 10 email addresses to send.'));
+  }
+}
+
+/**
+ * Callback builds a table of reports to email mailing.
+ *
+ * @return
+ *   Returns tabular representation of values results.
+ */
+function ding_dibs_email_reports() {
+  $time = REQUEST_TIME - 10800;
+  $report = db_select('dibs_transactions', 'd')
+    ->fields('d', array(
+      'payment_order_id',
+      'payment_transaction_id',
+      'payment_time',
+      'customer_uid',
+      'order_info_short',
+      'order_info_long',
+      'payment_price',
+      'payment_status',
+      'provider_payment_status',
+      'dibs_capture_status',
+    ))
+    ->condition('payment_time', date('Y-m-d H:i:s', $time), '>')
+    ->condition('d.api_module', 'ding_dibs')
+    ->condition('d.api_delta', 1)
+    ->orderBy('payment_order_id', 'DESC')
+    ->execute()
+    ->fetchAll();
+  $max_count = count($report);
+  $no_transactions_msg = '<p>' . t('No transactions found.') . '</p>';
+  if (!$max_count) {
+    return $no_transactions_msg;
+  }
+  for ($count = 0; $count < $max_count; $count++) {
+    if (!($report[$count]->payment_status == $report[$count]->provider_payment_status && $report[$count]->payment_status == $report[$count]->dibs_capture_status && $report[$count]->provider_payment_status == $report[$count]->dibs_capture_status)) {
+      $report_valid[] = $report[$count];
+    }
+  }
+  $rows = array();
+  drupal_add_css(drupal_get_path('module', 'ding_dibs') . '/css/ding_dibs.css');
+  $header = array(
+    array(
+      'data' => t('Order ID'),
+      'class' => 'payment',
+    ),
+    array(
+      'data' => t('Transaction ID'),
+      'class' => 'payment',
+    ),
+    array(
+      'data' => t('Date'),
+      'class' => 'payment',
+    ),
+    array(
+      'data' => t('User'),
+      'class' => 'payment',
+    ),
+    array(
+      'data' => t('Text'),
+      'class' => 'payment',
+    ),
+    array(
+      'data' => t('Amount'),
+      'class' => 'payment',
+    ),
+    array(
+      'data' => t('Cleared'),
+      'class' => 'payment',
+    ),
+    array(
+      'data' => t('Paid'),
+      'class' => 'payment',
+    ),
+    array(
+      'data' => t('Captured'),
+      'class' => 'payment',
+    ),
+    array(
+      'data' => '',
+      'class' => 'payment',
+    ),
+  );
+  foreach ($report_valid as $record) {
+    $status = $record->payment_status * 1 . $record->provider_payment_status * 1 . $record->dibs_capture_status * 1;
+    switch ($status) {
+      case '111':
+      case '000':
+        $status = 'ok';
+        break;
+
+      case '100':
+        $status = 'warning';
+        break;
+
+      default:
+        $status = 'error';
+    }
+    $order_info_long = unserialize($record->order_info_long);
```

Comment:
Please add a code comment explaining what is going on here and why. The extraction and manipulation of `$order_info_long` is not selfdocumenting.

Consider moving this into a separate function.